xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/meta.js

```diff
@@ -30,21 +30,51 @@ define(['jquery'], function($) {
     var documentReference = XWiki.Model.resolve(html.data('xwiki-reference'), XWiki.EntityType.DOCUMENT);
     var wikiReference     = documentReference.extractReference(XWiki.EntityType.WIKI);
     var spaceReference    = documentReference.extractReference(XWiki.EntityType.SPACE);
+
     return {
       'documentReference': documentReference,
-       // deprecated, use 'documentReference' instead
+      // deprecated, use 'documentReference' instead
       'document':          XWiki.Model.serialize(documentReference.relativeTo(wikiReference)),
-       // deprecated, use 'documentReference' instead
+      // deprecated, use 'documentReference' instead
       'wiki':              wikiReference.getName(),
-       // deprecated, use 'documentReference' instead
+      // deprecated, use 'documentReference' instead
       'space':             XWiki.Model.serialize(spaceReference.relativeTo(wikiReference)),
-       // deprecated, use 'documentReference' instead
+      // deprecated, use 'documentReference' instead
       'page':              documentReference.getName(),
       'version':           html.data('xwiki-version'),
       'restURL':           html.data('xwiki-rest-url'),
       'form_token':        html.data('xwiki-form-token'),
       // Since 10.4RC1
-      'userReference':     XWiki.Model.resolve(html.data('xwiki-user-reference'), XWiki.EntityType.DOCUMENT)
+      'userReference':     XWiki.Model.resolve(html.data('xwiki-user-reference'), XWiki.EntityType.DOCUMENT),
+      // Since 11.2RC1
+      'isNew': html.data('xwiki-isnew'),
+      'bumpVersion': function (isMinor) {
+        var version = html.data('xwiki-version');
+        var isNew = html.data('xwiki-isnew');
+
+        if (isNew.toString() === "true") {
+          version = "1.1";
+          html.data('xwiki-isnew', "false");
+        } else {
+          if (typeof version === "number") {
+            version = version.toString();
+          }
+          var parts = version.split('\.');
+          if (isMinor) {
+            parts[1] = parseInt(parts[1]) + 1;
+          } else {
+            parts[0] = parseInt(parts[0]) + 1;
+            parts[1] = 1;
+          }
+          version = parts[0] + "." + parts[1];
+        }
+        html.data("xwiki-version", version);
+        document.fire('xwiki:document:changeVersion', {'version': version, 'documentReference': documentReference});
+      },
+      'setVersion': function (version) {
+        html.data("xwiki-version", version);
+        document.fire('xwiki:document:changeVersion', {'version': version, 'documentReference': documentReference});
+      }
```

Comment:
Looks good. 

Since we do not really return a "map" kind of objects anymore, I think I would have created a "class" instead, so I could re-use the other methods (like `setVersion()` in `bumpVersion()`). With this solution, you would not need to update the DOM like you do. Instead you would change a field of the object and that's it, since nobody is supposed to rely on the DOM anymore.

Example of javascript "class" without using Prototype: https://github.com/xwiki/xwiki-platform/blob/87bce44d2b261389b565ece64ba1fbffbba01218/xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/AdminThemesSheet.xml#L151

Again, it does not mean what you did is wrong, but since you're asking for my feedbacks :)

Reply:
Thanks for the feedback! I wasn't actually quite happy of my solution and I thought about refactoring it as a real JS object. So ok I'll take time to do it.